storagetransfer/api/StorageTransfer.Samples.Tests/StorageFixture.cs

```diff
@@ -97,16 +168,47 @@ public void Dispose()
             }
             catch (Exception)
             {
-                // Do nothing, we delete on a best effort basis.
+                // If bucket is not empty, we delete on a best effort basis.
+                foreach (var storageObject in Storage.ListObjects(BucketNameSink, ""))
+                {
+                    Storage.DeleteObject(BucketNameSink, storageObject.Name);
+                }
+                Storage.DeleteBucket(BucketNameSink);
             }
             try
             {
                 Storage.DeleteBucket(BucketNameSource);
             }
             catch (Exception)
             {
-                // Do nothing, we delete on a best effort basis.
+                // If bucket is not empty, we delete on a best effort basis.
```

Comment:
And same here, when deleting the bucket, use the option for deleting a bucket and its contents.